packages/core/src/actions/smartAccount/internal/runMiddlewareStack.ts

```diff
@@ -51,7 +56,9 @@ export async function _runMiddlewareStack<
     client.middleware.feeEstimator,
     client.middleware.gasEstimator,
     client.middleware.customMiddleware,
-    client.middleware.paymasterAndData,
+    overrides && bypassPaymasterAndData(overrides)
+      ? bypassPaymasterMiddleware
+      : client.middleware.paymasterAndData,
```

Comment:
we shouldn't do this here. if we do, then none of the code you have written in the alchemy paymaster will even run. we should instead let paymaster middlewares handle paymaster overrides as they want to. some paymaster middlewares might be ok with overrides that are non `0x`

Reply:
that was the logic/reasoning behind making that change in the pr https://github.com/alchemyplatform/aa-sdk/pull/580 that you @moldy530 are suggesting here.

However, during testing, @dphilipson and @alex-miao gave a devEx feedback that for passing in '0x', the expectation would be to bypass the configured paymaster. That was why this was added.

cc: @avasisht23 @rthomare

Reply:
what I'm saying is that if you add this here then https://github.com/alchemyplatform/aa-sdk/pull/606/files#diff-cf46ef1442acb1802f8f34e60aea43e0ef05ce00d0078f2018ba2b1bf3dca895R151 will never be hit. If the idea is that passing in `0x` should skip all middleware then there's no need to add the changes to the `gasManager` middleware

Reply:
As a user I wouldn't expect that it would skip all middleware but rather it run all the middleware and just overriding the single overridden value at the end.

Reply:
Hm, I might actually expect it would skip the paymaster middleware. If I'm going to be overriding the paymaster information, I wouldn't expect anything of value to come out of running the paymaster middleware so doing so adds extra calls and waiting that I don't need.

On top of that, it's likely that the reason I'm overriding the paymaster is because the operation I'm sending is incompatible with the paymaster in the middleware, so if I do try to run the paymaster middleware on my op then the middleware would throw.

Reply:
(I agree with moldy's point about how adding the check here makes it unnecessary to add the check in the gas manager middleware)

Reply:
Oh I misread and thought that check was happening in the paymasterAndData middleware, not the overridden fee and gas estimators 👍 Yea the fee and gas estimator will run

Reply:
yes